src/JSON.jl

```diff
@@ -149,6 +149,12 @@ function _writejson(io::IO, state::State, s::AbstractString)
     Base.print(io, '"')
 end
 
+# workaround for issue in Julia 0.5.x where Float32 values are printed as
+# 3.4f-5 instead of 3.4e-5
+if v"0.5-" <= VERSION < v"0.6.0-dev.788"
```

Comment:
But I believe the relevant commit is https://github.com/JuliaLang/julia/commit/fc75f4d2e4e468aeabaf8fa948410cc44c056afe, which is 0.5.0-rc0+138.